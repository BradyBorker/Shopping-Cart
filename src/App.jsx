// import { useState } from 'react'
import { useParams } from "react-router-dom"
import Navbar from "./components/Navbar"
import Shop from "./components/Shop";
import Home from "./components/Home";
import "./styles/app.css"

function App() {
  const { page } = useParams();
  const isShopPage = page === 'shop';

  return (
    <>
      <Navbar />
      {isShopPage ? <Shop /> : <Home />}
    </>
  )
}

export default App
