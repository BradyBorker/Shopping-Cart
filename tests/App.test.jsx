import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { BrowserRouter } from 'react-router-dom';
import App from '../src/App';

describe('App component', () => {
    it('renders the root page', () => {
        const { container } = render(<BrowserRouter><App /></BrowserRouter>);
        expect(container).toMatchSnapshot();
    })

    it('renders the NavBar', () => {
        render(<BrowserRouter><App /></BrowserRouter>);

        expect(screen.getByRole('link', { name: 'Home' })).toHaveAttribute('href', '/');
        expect(screen.getByRole('link', { name: 'Shop' })).toHaveAttribute('href', '/shop');
        expect(screen.getByAltText('Shopping Cart').parentNode).toHaveAttribute('href', '/checkout')
    })

    it('renders the Home page first', () => {
        render(<BrowserRouter><App /></BrowserRouter>);

        expect(screen.getByAltText('Store')).toBeInTheDocument();
        expect(screen.getByRole('article')).toBeInTheDocument();
    })
})