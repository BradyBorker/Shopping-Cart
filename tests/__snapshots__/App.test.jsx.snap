// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders the root page 1`] = `
<div>
  <div
    class="_navbar_852229"
  >
    <a
      class="_link_852229"
      href="/"
    >
      Home
    </a>
    <div
      class="_shopLinks_852229"
    >
      <a
        class="_link_852229"
        href="/shop"
      >
        Shop
      </a>
      <a
        class="_checkoutLink_852229"
        href="/checkout"
      >
        <img
          alt="Shopping Cart"
          class="_svg_852229"
          src="https://www.svgrepo.com/show/522664/shopping-cart.svg"
        />
      </a>
    </div>
  </div>
  <div
    class="_home_3f2dd1"
  >
    <main
      class="_main_3f2dd1"
    >
      <img
        alt="Store"
        src="https://media.istockphoto.com/id/1275827044/vector/grocery-supermarket-interior-with-full-product-shelves-retail-and-consumerism-vector-concept.jpg?s=612x612&w=0&k=20&c=OtJhSNF6ya2VRmBBEn_dIjwVCq2h8PD7u9NRbcyE9jA="
      />
      <article
        class="_article_3f2dd1"
      >
        <h2
          class="_centerText_3f2dd1"
        >
          Lorem ipsum
        </h2>
        <p
          class="_fontSize_3f2dd1"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </article>
    </main>
  </div>
</div>
`;
